### ISSUE TITLE
Magento 2.2.6 and PHP 7.1 compatibility

### ISSUE THREAD

**Issue by falco-knapp** [Inxmail/inx_magento2#47]
The following patch has been provided to provide compatibility to Magento 2.2.6 and PHP 7.1.

[Magento2.2.6_PHP7.1_Kompatibilität_Config.zip](https://github.com/Inxmail/inx_magento2/files/2556691/Magento2.2.6_PHP7.1_Kompatibilitat_Config.zip)

**Comment by falco-knapp** [comment]
@bjoern-flagbit Could you please validate the patch and prepare a commit of it? Many thanks.

**Comment by bjoern-flagbit** [comment]
Hey Falco, we think this changes will just cover a PHP warning that is thrown when mapping is empty and a empty string is returned from config. But we suggest to handle this with the code above and just take care about the empty string.

*Changed - Model/Config/SystemConfig.php*
```
/**
 * @return array
 */
public function getMapConfig(): array
{
    if (empty($this->_data[self::CONFIG_FIELD_CUSTOMER_MAPPING])) {
        $this->_data[self::CONFIG_FIELD_CUSTOMER_MAPPING] = $this->_helper->getConfig(self::CONFIG_PATH_CUSTOMER_MAPPING);
    }

    if($this->_data[self::CONFIG_FIELD_CUSTOMER_MAPPING] === '') {
        return [];
    }

    return unserialize($this->_data[self::CONFIG_FIELD_CUSTOMER_MAPPING]) ?? [];
}
```
Can you ask your customer if this is working for him? And if not he should provide more informations about types and data that is coming in in his case.

**Comment by jwohlfeil** [comment]
Tested the module in 2.2.6 and PHP 7.1. When activating it in the configuration section it looks like that:   http://take.ms/v8LEj , even with the patch above applied.

**Comment by plel** [comment]
Hello,

i tried to debug that issue but since our old Inxmail dev credentials are not working anymore it is not possible to do so.

For the moment i think there is another reason why that error occurs. While i can not say for sure where that problem is my guess is that either the mapping data itself are not valid or there is another issue.

The Inxmail module (without configuration) works for me on a container with PHP 7.1.12 and M2 2.2.6.

**Comment by jwohlfeil** [comment]
Without configuration it works for me, too. But with configuration it doesn't :-)

**Comment by plel** [comment]
Like i thought the problem is on the mapping data. Since Magento did change the serializer the unserialize can not be used. It has to be a json_decode now.

The options are stored and recovered using Magento's ArraySerialized class on the backend. While Magento changed this class to use JSON at some point old data may not work on the loading (as still serialized using the php serialize stored in the database).
On the sync of subscribers the above mentioned code can not work anymore when there is a new configuration stored as then there will be a "false" result every time. That problem should only occur on the sync command however.

PR submitted with fix.

**Comment by falco-knapp** [comment]
I've released version 1.0.2.
